### ISSUE TITLE
Two issues with Lint/CircularArgumentReference cop

### ISSUE THREAD

**Issue by bquorning** [bbatsov/rubocop#2079]
Given this file `test.rb`:

```rb
# encoding: utf-8

# nodoc #
class A
  def self.value
    42
  end

  def puts_value(value: self.class.value)
    puts value
  end

  def foo(string:, length: string.length)
    puts string
    puts length
  end
end

A.new.puts_value
A.new.foo(string: 'hello')
```

I get these RuboCop offenses (using cdfb667):

```
» bin/rubocop test.rb
Inspecting 1 file
W

Offenses:

test.rb:9:25: W: Circular argument reference - value.
  def puts_value(value: self.class.value)
                        ^^^^^^^^^^^^^^^^
test.rb:13:28: W: Circular argument reference - length.
  def foo(string:, length: string.length)
                           ^^^^^^^^^^^^^

1 file inspected, 2 offenses detected
```

Running the file with `ruby -w` gives me no warnings:

```
» ruby -w test.rb
42
hello
5
» ruby -v
ruby 2.2.2p95 (2015-04-13 revision 50295) [x86_64-darwin13]
```

@maxjacobson

**Comment by maxjacobson** [comment]
Taking a look now

**Comment by maxjacobson** [comment]
And good catch!

**Comment by bquorning** [comment]
I wonder, is my 2nd example actually circular: `def foo(string:, length: string.length)`? Or is it allowed to call a method on one of the other keyword arguments?

**Comment by maxjacobson** [comment]
it should be allowed to call a method on one of the other keyword arguments
